docs: add cloud config docs (#2066)
apollographql/apollo-cli: docs/source/commands/cloud.mdx

---
title: Rover cloud Commands
subtitle: Manage cloud resources
description: Fetch and manage cloud resources with the rover cloud set of commands.
---

> Rover enables you to manage cloud resources. Currently, you can manage router config for a given graph ref. You manage your cloud resources with the `rover cloud` set of commands.

Reviewer: Do we want to be specific that's the Cloud Router config herE?

## Cloud Router Config

### `cloud config fetch`

The `cloud config fetch` attempts to fetch the current cloud router config for a given graph ref:

```
$ rover cloud config fetch <GRAPH_REF>
```

### `cloud config update`

> The `cloud config update` attempts to update a cloud router config for a given graph ref:

Reviewer: Do we want to point out you can use stdin to pass stuff in as well? Perhaps an example would help?

```
$ rover cloud config update --file <FILE> <GRAPH_REF>

Successfully updated cloud router config!
```

### `cloud config validate`

> The `cloud config validate` attempts to validate a cloud router config for a given graph ref:

Reviewer: Is it worth showing an example of the kind of errors we could get back?

---

> title: Rover cloud Commands

Reviewer 2: ```suggestion
title: Rover cloud commands
```

subtitle: Manage cloud resources

> description: Fetch and manage cloud resources with the rover cloud set of commands.

Reviewer 2: ```suggestion
description: Fetch and manage cloud resources with the `rover cloud` set of commands.
```

---

> Rover enables you to manage cloud resources. Currently, you can manage cloud router config for a given graph ref. You manage your cloud resources with the `rover cloud` set of commands.

Reviewer 2: ```suggestion
Rover enables you to manage cloud resources with the `rover cloud` set of commands. The commands allow you to manage cloud router configurations for a given graph ref.
```

## Cloud Router Config

### `cloud config fetch`

> The `cloud config fetch` attempts to fetch the current cloud router config for a given graph ref:

Reviewer 2: Can add example output for whether the attempt succeeds or fails

```
$ rover cloud config fetch <GRAPH_REF>
```

### `cloud config update`

The `cloud config update` attempts to update a cloud router config for a given graph ref:

```
$ rover cloud config update --file <FILE> <GRAPH_REF>

Successfully updated cloud router config!
```

### `cloud config validate`

The `cloud config validate` attempts to validate a cloud router config for a given graph ref:

```
$ rover cloud config validate --file <FILE> <GRAPH_REF>

No errors!
```

> Config that fails to validate will have the associated errors shown. For example, passing an invalid key/value pair `invalid: true`, as a `header`:

Reviewer 2: ```suggestion
An invalid config will have the associated errors shown. For example, passing an invalid key/value pair `invalid: true`, as a `header` produces the following:
```
